LG-15928 In-Person Proofing Passport Form

gina-yamada opened the pull request:
## 🎫 Ticket
[LG-15928](https://cm-jira.usa.gov/browse/LG-15928) Add passport form to ID-IPP flow (without form submit)

## 🛠 Summary of changes
* New IPP Passport view, controller, route
* New analytic event `idv_in_person_proofing_passport_visited` that logs on show
* New specs

Out of Scope
* Translations will be updated in another ticket 
* The field for month in the pr is an input. The mocks were updated to use a select drop down. Updating (or creating a new component) month for dob and passport expiration will be handled in a separate ticket.
* Form validation, form submission

## 📜 Testing Plan
Provide a checklist of steps to confirm the changes.

**Scenario 1: Global passports and In-Person Passports is enabled**
Setup: Ensure the following environment variables are set

```
  in_person_proofing_enabled: true
  in_person_proofing_opt_in_enabled: true
  in_person_passports_enabled: true
  doc_auth_passports_enabled: true
  doc_auth_passports_percent: 100
```

- [x] Login through the oidc sinatra application selecting the Identity Verified level of service.
- [x] Create a new account
- [x] Complete the ID-IPP flow reaching the post office selection page
- [x] Select a post office
- [x] Ensure user is navigated to the Choose your ID type page and select U.S. passport book
- [x] Ensure user is navigated to the IPP Passport view at url `/verify/in_person/passport`
- [x] Ensure the new aws analytic event for IPP Passport visited logs
- [x] Ensure the [Figma](https://www.figma.com/design/RFPjGWYkR6bkBWl9QpKUCL/U.S.-passports-in-ID-IPP?node-id=1177-6562&t=bjsetx7f8b1KflGR-0) mocks match the view. Inspect the entire page ie: step navigation, header, info below header, each field name, helper text, buttons, having trouble section, links
- [x] Ensure the form is responsive
- [x] Ensure the form renders for each language (English, Spanish, French, Chinese). The translations are not available so all content will be in English.
- [x] Ensure the Use another type of ID link takes you to the IPP Choose your ID type page
- [x] Ensure Cancel is working as expected (A. Start over takes you to welcome B. No, Keep going kicks you back to IPP Passports)
- [x] On Welcome [and/or anytime before an establish enrollment is created], change the url by typing /verify/in_person/passport. Ensure you are not taken to the passport view. An establishing enrollment has not been created so you should not be


**Testing steps for Scenario 2-4**
- [x] Login through the oidc sinatra application selecting the Identity Verified level of service.
- [x] Create a new account
- [x] Complete the ID-IPP flow reaching the post office selection page
- [x] Select a post office
- [x] Ensure user is navigated to the state ID page
- [x] Change the url by typing /verify/in_person/passport. Ensure you are not taken to the passport view


**Scenario 2: Global passports is enabled and In-Person Passports is disabled**
Setup: Ensure the following environment variables are set
Run through testing steps listed above

```
  in_person_proofing_enabled: true
  in_person_proofing_opt_in_enabled: true
  in_person_passports_enabled: false
  doc_auth_passports_enabled: true
  doc_auth_passports_percent: 100
```

**Scenario 3: Global passports is disabled and In-Person Passports is enabled**
Setup: Ensure the following environment variables are set
Run through testing steps listed above

```
  in_person_proofing_enabled: true
  in_person_proofing_opt_in_enabled: true
  in_person_passports_enabled: true
  doc_auth_passports_enabled: false
  doc_auth_passports_percent: 100
```

**Scenario 4: Global passports and In-Person Passports is disabled**
Setup: Ensure the following environment variables are set
Run through testing steps listed above

```
  in_person_proofing_enabled: true
  in_person_proofing_opt_in_enabled: true
  in_person_passports_enabled: false
  doc_auth_passports_enabled: false
  doc_auth_passports_percent: 100
```


## 👀 Screenshots

<summary>New IPP Passport Page</summary>

**Image taken on 05/01**

![Screenshot 2025-05-01 at 7 49 36 AM](https://github.com/user-attachments/assets/55d3ae7c-0238-4380-a4de-ced7e90b708f)

</details>

gina-yamada commented:
@rutvigupta-design  I included screenshots above of the new IPP Passport view above. If you could take a look and see if it looks pixel perfect to the mocks. Thank you

rutvigupta-design commented:
@gina-yamada I think mostly everything looks great to me! Just two quick things: 

- Can you please confirm that the spacing between the Expiration date input field and the 'Continue' button is 40px? 
- For the "Use a driver's license or state ID" troubleshooting link at the bottom, I think we won't want an external link icon since this is directing users within the flow

gina-yamada commented:
> @gina-yamada I think mostly everything looks great to me! 🎉 Just two quick things:
> 
> * Can you please confirm that the spacing between the Expiration date input field and the 'Continue' button is 40px?
> * For the "Use a driver's license or state ID" troubleshooting link at the bottom, I think we won't want an external link icon since this is directing users within the flow

@rutvigupta-design  Good eye! The spacing between the Expiration date input field and the Continue button is now 40px. It was 32px before. The external link icon is now removed. See screenshots below.

![Screenshot 2025-04-29 at 7 52 29 AM](https://github.com/user-attachments/assets/599ab720-4e70-44f7-be96-442f6392e9f2)

![Screenshot 2025-04-29 at 7 58 19 AM](https://github.com/user-attachments/assets/f6d34c71-b10d-4485-b532-b6224ff3f40b)

![Screenshot 2025-04-29 at 8 00 40 AM](https://github.com/user-attachments/assets/340256c2-c886-47f3-8a7c-6b56ff040adc)

gina-yamada commented:
@rutvigupta-design  I updated the UX and spec to display Submit on the button of the the Passport form rather than Continue. I updated the screenshot in the description at the very top of this pull request.

rutvigupta-design commented:
I just re-reviewed this and this LGTM! Approved from UX, thank you for making the fixes so quick!! 🎉

gina-yamada commented on app/controllers/idv/in_person/passport_controller.rb:
```diff
@@ -1,6 +1,63 @@
+          controller: self,
+          next_steps: [:ipp_address],
+          preconditions: ->(idv_session:, user:) {
+            idv_session.in_person_passports_allowed? && user.has_establishing_in_person_enrollment?
```
I thought about adding this additional check:

```
&& DocumentCaptureSession.find_by(uuid: idv_session.document_capture_session_uuid).passport_status == 'requested'
```

> This check ensures that the user came from the Choose your ID type and clicked on their choice. The benefit of adding it to preconditions is that a user will not be able to navigate to /verify/in_person/passports by typing in to the URL after an establishing enrollment has created (so basically once on the Choose your ID screen).  The likely hood of user changing the URL by tying it is low.

> If I added this check, it will cause trouble later if we add a Help section on State ID that allows a user to come directly to try passports without going back to Choose your ID. (That is where this value is being set. We would have to add additional logic to set it.) We are not checking this on State ID. Additionally, the way I understand this- it is if the user can attempt with Passport- not if they did.

I believe we do not need to add this check but wanted to call it out.

shanechesnutt-ft replied:
I think we can add the check here to ensure that passport has been selected. I don't think that state_id needs the check.

gina-yamada replied:
I worked with Rutvi, Shane, and Annie. The help link at the bottom of the page now goes back to Choose your ID so this additional check makes sense. You will always be coming from the Choose your ID page. If you came from remote and they decide to skip the IPP Choose your ID page, this value would still be set.

shanechesnutt-ft reviewed (commented):
Nice work Gina! We walked through this yesterday and you address everything except for adding the check for passport status requested in the preconditions. Once that is added I will approve.

shanechesnutt-ft commented on config/locales/en.yml:
```diff
@@ -1296,6 +1297,16 @@ in_person_proofing.body.state_id.alert_text: Passports are not accepted in perso
+in_person_proofing.form.passport.first_name_hint: Do not include your middle name if you have one listed under ‘Given names’
+in_person_proofing.form.passport.passport_number: Passport number
+in_person_proofing.form.passport.passport_number_hint: 'Your passport number is 9 characters long. Refer to the top right corner of your passport book.'
+in_person_proofing.form.passport.redirect_to_state_id: Use a driver’s license or state ID instead
```
Is this needed anymore?

gina-yamada replied:
Thank you! The linter caught it too. It is now gone.

gina-yamada commented on app/views/idv/in_person/passport/show.html.erb:
```diff
@@ -0,0 +1,91 @@
+           url: idv_in_person_choose_id_type_url,
+           new_tab: false,
+         ).with_content(t('idv.troubleshooting.options.use_another_id_type')) %>
+<% end %>
```
This [Figma](https://www.figma.com/design/RFPjGWYkR6bkBWl9QpKUCL/U.S.-passports-in-ID-IPP?node-id=802-10368&t=CumEvH09KRUlkCgG-0) reflects the most recent version of Passport form. The help content displays the same content as remote (Use another type of ID) and link you back to the Choose your ID type. IPP will go to the IPP version of that view.

rutvigupta-design commented:
Updated screenshot and AC LGTM! 🎉

shanechesnutt-ft reviewed (approved):
I ran through the test plan and things look great! Nice job! 👍🏻

WilliamBirdsall reviewed (approved):
Ran through testing steps. Things look good!

Great work on this one!!

lmgeorge reviewed (approved):
LGTM! Caveat: I was only able to do a read through due to device issues. My only requested change is doc comment and commes with a suggestion so hopefully it will be easy to apply.

Great work!

gina-yamada commented on app/services/analytics_events.rb:
```diff
@@ -3241,6 +3241,27 @@ def idv_in_person_proofing_nontransliterable_characters_submitted(
+  # @param [String] analytics_id Current IdV flow identifier
+  # @param [Boolean] opted_in_to_in_person_proofing User opted into in person proofing
+  # @param [Boolean] skip_hybrid_handoff Whether skipped hybrid handoff A/B test is active
+  def idv_in_person_proofing_passport_visited(
```
Done. See commit 9ca3a95521a7f69bd0f6b10d090a2cd317ad1aa2
